docs/adr/0039-locales-languages.md

```diff
@@ -0,0 +1,183 @@
+
+### État du code de Pix
+
+Actuellement dans le code de Pix les 2 notions et noms de *locale* et *language* sont manipulées :
+
+* la notion et le nom de *language* sont utilisés pour le côté client
+* la notion et le nom de *locale* sont utilisés côté BDD et hapi
+
+Et en ce qui concerne le format des identifiants de *language*/*locale*, c'est le format suivant qui est utilisé : `fr`, `fr-fr`, `fr-be`. Ce qui n'est donc pas la forme canonique de représentation suivant la spécification *BCP 47*.
+
+
+## Solution proposée : Fonder tous les traitements sur la notion de locales
+
+**Description**
+
+Dans le cas de Pix c'est la notion de *locale* qui est la plus importante car une *locale* est une association langue + autres données liées au groupe auquel on s'adresse, ce qui correspond bien aux cas d'utilisation de Pix où chaque utilisateur fait partie d'un groupe linguistique-géolocalisé, même si la géolocalisation de ce groupe peut parfois être très large.
+
+On souhaite donc :
+
+* associer à chaque utilisateur une *locale* (et non plus un *language*) au format BCP 47 avec persistance de ce choix en base de données
+* avoir les différentes versions des sites web identifiées par une *locale* au format BCP 47 et non plus des identifiants non-normés
+* Utiliser le builtin JavaScript `Intl.Locale` chaque fois que c'est possible
+
+**Avantage(s) :**
+
+- Traiter systématiquement des *locales* est une recommandation facile à comprendre et à suivre
+- Le builtin JavaScript `Intl.Locale` est un outil disponible. Même si `Intl.Locale` n'est pas disponible sur tous les navigateurs on pourrait ne l'utiliser d'abord que côté serveur et utiliser sa forme sérialisée par la méthode `toString` partout ailleurs.
+
+**Inconvénient(s) :**
+
+- Actuellement la table `users` a une colonne `lang` et non `locale`, ce qui demanderait du travail de migration de données et de code si on veut passer à `locale`
+
+## Décision
+
+* Recueillir les avis et commentaires de tous les développeurs Pix
+* Mettre en oeuvre cet ADR dans *pix-site*, le faire évoluer si besoin en fonction de l'expérience acquise
+
+   Concrètement :
+
+   * On ne change pas l'identifiant de la locale `fr` déjà existant car c'est un identifiant de locale valide. Cette locale propose une version internationale sans cibler une région particulière.
+   * On changera l'identifiant `fr-fr`, qui n'est pas un identifiant recommandé (car la région doit être en majuscules), en `fr-FR`.
+   * On changera `en-us` et `en-gb`, qui ne sont pas des identifiants recommandés (car la région doit être en majuscules), en simplement `en` car on souhaite proposer une version en anglais de manière internationale, sans cibler cette version pour une région particulière.
```

Comment:
À valider en interne, pour l'instant ce point est assez peu clair

Reply:
@yannbertrand là aussi l'objet de cet ADR de normaliser, mais c'est peut-être à faire dans une itération ultérieure.

Reply:
Je suis plutôt pour supprimer ce point dans ce cas. Pour [pix-site effectivement on fera ça](https://1024pix.slack.com/archives/CKQ5FDE3H/p1670944341263589), mais dans le référentiel on stockera probablement un jour ces 2/3 formats.

Reply:
Suite à discussion, j'avais perdu le contexte de ces éléments qui référent à une feuille de route sur les pix-site. Ce point sera effectivement traité dans notre chantier concret du moment, donc laissons le 👍

Reply:
Le commit 442bdcbb4bd5a4ac6647ea7eee637f93b3ad8b9a rend plus clair le cycle de vie de cet ADR, en proposant notamment de centraliser les éventuels avis et commentaires dans un ticket Jira externe (@yannbertrand et non dans une issue GitHub comme lors de notre discussion car, à la réflexion, seules les discussions purement techniques liées à une PR ont lieu sur GitHub).